pkg/client/ghcr/path.go

```diff
@@ -1,16 +1,27 @@
 package ghcr
 
 import (
+	"regexp"
 	"strings"
 )
 
+const (
+	HostRegTempl = `^(containers\.[a-zA-Z0-9-]+\.ghe\.com|ghcr\.io)$`
+)
+
+var HostReg = regexp.MustCompile(HostRegTempl)
+
 func (c *Client) IsHost(host string) bool {
 	// Package API requires Authentication
 	// This forces the Client to use the fallback method
 	if c.opts.Token == "" {
 		return false
 	}
-	return host == "ghcr.io"
+	// If we're using a custom hostname.
+	if c.opts.Hostname != "" && c.opts.Hostname == host {
```

Comment:
do you need to check if `c.opts.Hostname` is empty _and_ equal to `host`, or just `== host`?